zgw/catalogi-api/src/main/kotlin/com/ritense/catalogiapi/CatalogiApiPlugin.kt

```diff
@@ -121,195 +139,246 @@ class CatalogiApiPlugin(
+            CATALOGI_API.STATUSTYPE to zaakTypeUrl.toString(),
+            CATALOGI_API.STATUSTYPE to omschrijving
+        ) {
+            logger.debug { "Getting Statustype by omschrijving: $omschrijving for zaaktype $zaakTypeUrl" }
+            return getStatustypen(zaakTypeUrl)
+                .singleOrNull { it.omschrijving.equals(omschrijving, ignoreCase = true) }
+                ?: throw StatustypeNotFoundException("With 'omschrijving': '$omschrijving'")
+        }
     }
 
     fun getResultaattypen(zaakTypeUrl: URI): List<Resultaattype> {
-        var currentPage = 1
-        var currentResults: Page<Resultaattype>?
-        val results = mutableListOf<Resultaattype>()
-
-        do {
-            logger.debug { "Getting page of resultaattypen, page $currentPage for zaaktype $zaakTypeUrl" }
-            currentResults = client.getResultaattypen(
-                authenticationPluginConfiguration,
-                url,
-                ResultaattypeRequest(
-                    zaaktype = zaakTypeUrl,
-                    page = currentPage++
+        withLoggingContext(CATALOGI_API.RESULTAATTYPE to zaakTypeUrl.toString()) {
+            var currentPage = 1
+            var currentResults: Page<Resultaattype>?
+            val results = mutableListOf<Resultaattype>()
+
+            do {
+                logger.debug { "Getting page $currentPage of resultaattypen for zaaktype $zaakTypeUrl" }
+                currentResults = client.getResultaattypen(
+                    authenticationPluginConfiguration,
+                    url,
+                    ResultaattypeRequest(
+                        zaaktype = zaakTypeUrl,
+                        page = currentPage++
+                    )
                 )
-            )
-            results.addAll(currentResults.results)
-        } while (currentResults?.next != null)
+                results.addAll(currentResults.results)
+            } while (currentResults?.next != null)
 
-        return results
+            return results
+        }
     }
 
     fun getResultaattype(resultaatTypeUrl: URI): Resultaattype {
-        return client.getResultaattype(authenticationPluginConfiguration, url, resultaatTypeUrl)
+        withLoggingContext(CATALOGI_API.RESULTAATTYPE to resultaatTypeUrl.toString()) {
+            logger.debug { "Getting Resultaattype for URL: $resultaatTypeUrl" }
+            return client.getResultaattype(authenticationPluginConfiguration, url, resultaatTypeUrl)
+        }
     }
 
     fun getResultaattypeByOmschrijving(zaakTypeUrl: URI, omschrijving: String): Resultaattype {
-        return getResultaattypen(zaakTypeUrl)
-            .singleOrNull { it.omschrijving.equals(omschrijving, ignoreCase = true) }
-            ?: throw ResultaattypeNotFoundException("With 'omschrijving': '$omschrijving'")
+        withLoggingContext(
+            CATALOGI_API.RESULTAATTYPE to zaakTypeUrl.toString(),
+            CATALOGI_API.RESULTAATTYPE to omschrijving
```

Comment:
Both keys are the same, but i think we don't need the omschrijving as MDC variable.